app/classifier/tasks/shortcut-editor.jsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import AutoSave from '../../components/auto-save';
+import handleInputChange from '../../lib/handle-input-change';
+import Shortcut from './shortcut';
+
+export default class ShortcutEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.addAnswer = this.addAnswer.bind(this);
+    this.toggleShortcut = this.toggleShortcut.bind(this);
+  }
+
+  toggleShortcut(e) {
+    let nextTaskID;
+
+    if (e.target.checked) {
+      const taskCount = Object.keys(this.props.workflow.tasks).length;
+
+      let taskIDNumber = -1;
+
+      while (!((nextTaskID != null) && !(nextTaskID in this.props.workflow.tasks))) {
+        taskIDNumber += 1;
+        nextTaskID = `T${taskCount + taskIDNumber}`;
```

Comment:
Lines 17-23 look to be identical to `addNewTask` in the workflow lab page: https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/pages/lab/workflow.cjsx#L525-L541

Is there a reason why you're not just passing `addNewTask` down as a prop and using that? You can rewrite your handler to call `addNewTask` as well as the couple of Shortcut specific lines, I think just line 29 and line 31?